pilot/pkg/config/kube/gateway/conversion.go

```diff
@@ -2179,14 +2179,16 @@ func reportGatewayStatus(
 	obj.Status.(*kstatus.WrappedStatus).Mutate(func(s config.Status) config.Status {
 		gs := s.(*k8s.GatewayStatus)
 		addressesToReport := external
-		addrType := k8s.IPAddressType
 		if len(addressesToReport) == 0 {
+			wantAddressType := classInfo.addressType
+			if override, ok := obj.Annotations[addressTypeOverride]; ok {
```

Comment:
@howardjohn Doesn't Gateway API provide a mechanism to indicate address class / infra ? 

I'm not sure we should define a new Istio API ( annotations are APIs ) for this. 

Also not sure this is the right pattern - GKE for example has a different class for 'internal' gateways vs
external gateways. Given that generic OPA/etc can be used to control access based on gateway class (i.e. regular users allowed to create internal classes ) - it is better to be consistent with this pattern instead of having an 'external gateway' be turned into internal by a vendor-specific annotation.